components/monitoring/base/install_prometheus.yaml

```diff
@@ -0,0 +1,28 @@
+apiVersion: template.openshift.io/v1
+kind: Template
+metadata:
+  name: install-prometheus-operator
+objects:
+- apiVersion: operators.coreos.com/v1
+  kind: OperatorGroup
+  metadata:
+    name: openshift-customer-monitoring
+    namespace: ${MONITORING_NAMESPACE}
+  spec:
+    targetNamespaces:
+    - ${MONITORING_NAMESPACE}
+- apiVersion: operators.coreos.com/v1alpha1
+  kind: Subscription
+  metadata:
+    name: prometheus
+    namespace: ${MONITORING_NAMESPACE}
+  spec:
+    channel: beta
+    installPlanApproval: Automatic
+    name: prometheus
+    source: community-operators
+    sourceNamespace: openshift-marketplace
+    startingCSV: prometheusoperator.0.56.3
+parameters:
+- name: MONITORING_NAMESPACE
+  value: workload-monitoring
```

Comment:
I thought the namespace would be `openshift-user-workload-monitoring`?

Reply:
I did not want to put openshift here. because there are a lot of default openshift namespaces there which are mostly from the openshifts own components. but yes it is fine to use `openshift-user-workload-monitoring`

Reply:
Argh, this note from the documentation about `openshift-user-workload-monitoring` (https://docs.openshift.com/container-platform/4.6/monitoring/enabling-monitoring-for-user-defined-projects.html) concerns me:

`You must have access to the cluster as a user with the cluster-admin role to enable monitoring for user-defined projects in OpenShift Container Platform. Cluster administrators can then optionally grant users permission to configure the components that are responsible for monitoring user-defined projects.`

And this is even more concerning:
https://docs.openshift.com/container-platform/4.6/monitoring/managing-metrics.html#managing-metrics 

`Only cluster administrators have access to the third-party UIs provided with OpenShift Container Platform Monitoring.`
`Developers can only use the Developer perspective and not the Administrator perspective. As a developer, you can only query metrics for one project at a time. Developers cannot access the third-party UIs provided with OpenShift Container Platform monitoring that are for core platform components. Instead, use the Metrics UI for your user-defined project.`

It sounds like it might be easier to grant our developers the access they need if we have our own Prometheus instance in our own namespace.  But what do you all think?

Reply:
so what is the bes suggestion to use. I have added this to kustomization so that we can change in one place

Reply:
> You must have access to the cluster as a user with the cluster-admin role to enable monitoring for user-defined projects in OpenShift Container Platform. Cluster administrators can then optionally grant users permission to configure the components that are responsible for monitoring user-defined projects.

My understanding is that cluster-admin privileges are needed when we want to create the `cluster-monitoring-config` ConfigMap in the `openshift-monitoring`, which will trigger the deployment of a second Prometheus instance in the `openshift-user-workload-monitoring` namespace. We may not need such privileges if we want to deploy Prometheus ourselves, but if using the `openshift-user-workload-monitoring` namespace causes troubles, then I guess that another (non `openshift-*`) namespace would be fine, too.

Reply:
that being said, would it make sense to use something like `appstudio-`, `redhat-` or a combination of both as a prefix for such a namespace, if we decide not to use `openshift-user-workload-monitoring`? 
It's easier to identify and we could even have a rule (implemented by a webhook for example) to prevent users from creating namespaces with such a prefix.

Reply:
Let's not use the `redhat-` prefix - namespaces with this prefix have the same limitation as with `openshift-`. IIRC, you cannot even create such a namespace with dedicated-admin privileges.
So, if you want to deploy it to another namespace, then let's use the `appstudio-` prefix for that.

> It's easier to identify and we could even have a rule (implemented by a webhook for example) to prevent users from creating namespaces with such a prefix.

that's not needed - all namespaces in the cluster are created either via infra-deployments repo or via kcp syncer. There is no risk of facing any namespace name conflicts.

Anyway, I'll leave the decision about the location up to @xcoulon :wink:

Reply:
ah right, I forgot about the `redhat-` prefix which is also reserved!

Reply:
lets use `appstudio-workload-monitoring`

Reply:
ok, let's use that for now. Anyways, it should be easily changed with Kustomize